Enhanced login, sign up and forgot password pages

miyabitanimchi opened the pull request:
What I did on this branch is as follows:

- Added validation on login , sign up and forgot password pages (now error messages are shown on client too)
- Implemented a function to redirect to the browsing page (`/board`)
- Changed some style Kaho mentioned [here](https://github.com/shiv-chan/senpai-kohai/pull/5#pullrequestreview-811751367)
- some minor change, addition and deletion

Please check when you have time, thank you!

Dannyboy222 reviewed (approved):
👍

shiv-chan commented on client/src/features/signup/SignUp.tsx:
```diff
@@ -39,50 +87,58 @@ const SignUp = () => {
-                  setInputs({ ...inputs, email: e.target.value })
-                }
+                className="h-10 text-xl w-full pl-8 rounded"
+                onChange={(e) => handleOnChange(e)}
+                onBlur={(e) => handleOnBlur(e)}
```
```suggestion
                onChange={handleOnChange}
                onBlur={handleOnBlur}
```
We can make it shorter like this. The same goes for line 130-131.

shiv-chan commented on client/src/features/signup/SignUp.tsx:
```diff
@@ -1,36 +1,84 @@
+
+  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    errMsgFromServer.length !== 0 && setErrMsgFromServer('');
```
@miyabitanimchi I liked this 👀 💡

shiv-chan commented on client/src/features/login/LogIn.tsx:
```diff
@@ -8,57 +8,71 @@ import { FaUnlockAlt } from 'react-icons/fa';
-                  setInputs({ ...inputs, email: e.target.value })
-                }
+                className="mb-8 h-10 text-xl w-full pl-8 rounded "
+                onChange={(e) => handleOnChange(e)}
```
```suggestion
                onChange={handleOnChange}
```
The same goes for line 87 and 132.

shiv-chan commented on client/src/features/login/LogIn.tsx:
```diff
@@ -67,21 +81,20 @@ const LogIn = () => {
-                onChange={(e) =>
-                  setInputs({ ...inputs, password: e.target.value })
-                }
+                className="h-10 mb-12 w-full px-8 rounded"
```
maybe it's better to have the same font size as email input.

miyabitanimchi replied:
Thank you, I added text-xl for all the password inputs (`/login`, `/signup` and `forgotpassword/reset/:id`). 👍

shiv-chan commented on client/src/features/forgotpassword/ForgotPasswordResetSuccess.tsx:
```diff
@@ -0,0 +1,21 @@
+          New password set successfully!
+        </p>
+        <Link to="/login" className="flex-initial underline m-auto">
+          Login
```
**Suggestion:** Back to login

shiv-chan reviewed (changes requested):
Thank you for improving, @miyabitanimchi 🙏 

There's one thing I notice.
When you click "Send" button to reset a password with an empty input, the error message looks like the capture.
![スクリーンショット 2021-12-09 22 19 01](https://user-images.githubusercontent.com/51708229/145527494-b8ca5560-3667-4553-978e-ce2a891cbbfa.png)
Could you add the message for this case before merging?

Please check other comments too.

Thanks!

miyabitanimchi commented:
I improved code Kaho mentioned with the comments, and also added another error message when a user submit with empty value on `/forgotpassword` page. I added an error handling for that on `server/controllers/forgotPassword/forgotPassword.js`. 
Please review it again when you have time, and also let me know if I miss something and if you notice something 🙏

shiv-chan reviewed (approved):
It looks good to me ✨ 
Thanks for improving, @miyabitanimchi!

shiv-chan commented:
@miyabitanimchi Go ahead and merge this to the target branch.
I updated some wordings. Please refer to [this commit](https://github.com/shiv-chan/senpai-kohai/pull/21/commits/11483cf6d1737ca1a7cc38a9234f13c1286b07c5).

Thanks!
